rasa/nlu/test.py

```diff
@@ -1246,54 +1245,44 @@ def get_eval_data(
 
     intent_results, entity_results, response_selection_results = [], [], []
 
-    response_labels = [
+    response_labels = {
         e.get(INTENT_RESPONSE_KEY)
         for e in test_data.intent_examples
         if e.get(INTENT_RESPONSE_KEY) is not None
-    ]
+    }
     intent_labels = [e.get(INTENT) for e in test_data.intent_examples]
-    should_eval_intents = (
-        is_intent_classifier_present(interpreter) and len(set(intent_labels)) >= 2
-    )
-    should_eval_response_selection = (
-        is_response_selector_present(interpreter) and len(set(response_labels)) >= 2
-    )
-    available_response_selector_types = get_available_response_selector_types(
-        interpreter
-    )
-
-    should_eval_entities = (
-        is_entity_extractor_present(interpreter) and len(test_data.entities) > 0
-    )
+    should_eval_intents = len(set(intent_labels)) >= 2
+    should_eval_response_selection = len(response_labels) >= 2
+    should_eval_entities = len(test_data.entity_examples) > 0
 
     for example in tqdm(test_data.nlu_examples):
         result = interpreter.parse(example.get(TEXT), only_output_properties=False)
-
+        remove_entities_of_extractors(result, PRETRAINED_EXTRACTORS)
```

Comment:
optional: can this be a protected function?
```suggestion
        _remove_entities_of_extractors(result, PRETRAINED_EXTRACTORS)
```